src/account/UpgradeableModularAccount.sol

```diff
@@ -573,6 +609,41 @@ contract UpgradeableModularAccount is
     // solhint-disable-next-line no-empty-blocks
     function _authorizeUpgrade(address newImplementation) internal override {}
 
+    function _checkIfValidationApplies(bytes4 selector, FunctionReference validationFunction, bool isDefault)
+        internal
+        view
+    {
+        AccountStorage storage _storage = getAccountStorage();
+
+        // Check that the provided validation function is applicable to the selector
+        if (isDefault) {
+            if (
+                !_defaultValidationAllowed(selector)
+                    || !_storage.defaultValidations.contains(toSetValue(validationFunction))
+            ) {
+                revert UserOpValidationFunctionMissing(selector);
+            }
+        } else {
+            // Not default validation, but per-selector
+            if (!getAccountStorage().selectorData[selector].validations.contains(toSetValue(validationFunction))) {
+                revert UserOpValidationFunctionMissing(selector);
+            }
+        }
+    }
+
+    function _defaultValidationAllowed(bytes4 selector) internal view returns (bool) {
+        if (
```

Comment:
Maybe add comments to indicate that the following state-changing native functions are by default support default validations.

Reply:
Sounds good, added